src/usecases/current-figma-user-use-case.ts

```diff
@@ -1,12 +1,12 @@
-import type { ConnectInstallation } from '../domain/entities';
+import type { ConnectInstallation, FigmaUser } from '../domain/entities';
 import { figmaService } from '../infrastructure/figma';
 
-export const checkUserFigmaAuthUseCase = {
+export const currentFigmaUserUseCase = {
```

Comment:
**suggestion:** Please, consider creating a separate use case instead of modifying the existing one. Here is why it can be a good idea:

- Separation will allow to scale 2 business use case independently (the same as the corresponding endpoints) and reduce the risk of breaking one while changing another one. While these use cases look similar, they are are of a different nature (one is driven by admin UI, another -- by Jira) and already have differences.
- It will cause only minor code duplication since the majority of the logic is encapsulated in the service layer. These use cases are very thin.
- It is a common good approach to have a separate use case per endpoint since an endpoint is usually represent an entry point to some business operation.

This being the case, consider the following:

- Keep the existing `/auth/checkAuth` endpoint, which calls `checkUserFigmaAuthUseCase`.
- Rename the existing `/admin/auth/checkAuth` endpoint to `/admin/auth/me`, which calls _new_ `getCurrentFigmaUserUseCase`.

Please, let me know if you have any questions or concerns.

Reply:
sgtm!